lib/services/schema-object-factory.ts

```diff
@@ -300,21 +300,25 @@ export class SchemaObjectFactory {
     const enumName = metadata.enumName;
     const $ref = getSchemaPath(enumName);
 
-    if (!(enumName in schemas)) {
-      const enumType: string = (
-        metadata.isArray
-        ? metadata.items['type']
-        : metadata.type
-      ) ?? 'string';
-
-      schemas[enumName] = {
-        type: enumType,
-        enum:
-          metadata.isArray && metadata.items
-            ? metadata.items['enum']
-            : metadata.enum
-      };
-    }
+    // Allow given fields to be part of the referenced enum schema
+    const additionalParams = ['description', 'deprecated', 'default']
```

Comment:
I was not sure if we should just allow all fields, so I added an allowlist here. I did not find any information in the swagger spec on what they technically support on an enum ref field.